src/action.c

```diff
@@ -889,10 +901,20 @@ actions_run(struct view *activator, struct server *server,
 				view_move_to_edge(view, edge, snap_to_windows);
 			}
 			break;
+		case ACTION_TYPE_TOGGLE_SNAP_TO_EDGE:
 		case ACTION_TYPE_SNAP_TO_EDGE:
 			if (view) {
 				/* Config parsing makes sure that direction is a valid direction */
 				enum view_edge edge = action_get_int(action, "direction", 0);
+				if (action->type == ACTION_TYPE_TOGGLE_SNAP_TO_EDGE
+						&& !view->maximized
```

Comment:
nit: let's use `view->maximized != VIEW_AXIS_NONE` to keep consistency with other parts of the codebase.

Reply:
Actually I would prefer to keep `!view->maximized` to have a consistent toggle-tiling-back-from-maximed-state for all maximized variations.

Reply:
Sorry I should've said `view->maximized == VIEW_AXIS_NONE`. I'm so bad at understanding logic.

Well, I prefer `view->maximized == VIEW_AXIS_NONE` (which is exactly the same as `!view->maximized`) since it makes it clear that the view is not maximized in any direction.

> have a consistent toggle-tiling-back-from-maximed-state for all maximized variations.

Sorry I couldn't understand what you mean by this (I guess this is my English skill issue).

Reply:
Nah, I'm even worse at simply reading, somehow I read `VIEW_AXIS_BOTH` while you wrote `VIEW_AXIS_NONE` ;). I've pushed that adjustments and agree that consistency matters.